pvlib/shading.py

```diff
@@ -553,3 +553,101 @@ def shaded_fraction1d(
+    irradiance of non-monolithic Silicon
+    modules and arrays with an arbitrary number of bypass diodes.
+
+    .. versionadded:: 0.11.0
+
+    Parameters
+    ----------
+    shaded_fraction : numeric
+        Fraction of module surface area that is shaded. Unitless.
+    shaded_blocks : numeric
+        Number of blocks affected by the shadow. Unitless.
+        If a floating point number is provided, it will be rounded up.
+    total_blocks : int
+        Number of total blocks. Unitless.
+
+    Returns
+    -------
+    shading_correction_factor : numeric
+        Multiply direct and circumsolar irradiance by this factor.
+
+    Notes
+    -----
+    The implemented equation is (6) from [1]_:
+
+    .. math::
+
+        (1 - F_{ES}) = (1 - F_{GS}) (1 - \frac{N_{SB}}{N_{TB} + 1})
+
+    Where :math:`(1 - F_{ES})` is the correction factor to be multiplied by
+    the direct and circumsolar irradiance, :math:`F_{GS}` is the shaded
+    fraction of the collector, :math:`N_{SB}` is the number of shaded blocks
+    and :math:`N_{TB}` is the number of total blocks.
+
+    Blocks terminology
+    ^^^^^^^^^^^^^^^^^^
+    [1]_ defines a *block* as a group of solar cells protected by a bypass
+    diode. Also, a *block* is shaded when at least one of its cells is shaded.
+
+    How many blocks and their layout depend on the module(s) used. Many
+    manufacturers don't specify this information explicitly. However, we can
+    infer these values from:
+
+    - the number of bypass diodes
+    - where and how many junction boxes are present on the back of the module
+    - whether or not the module is comprised of *half-cut cells*
+
+    The latter two are heavily correlated.
+
+    For example:
+
+    1. A module with 1 bypass diode behaves as 1 block.
+    2. A module with 3 bypass diodes and 1 junction box is likely to have 3
+       blocks.
+    3. A half-cut module with 3 junction boxes (split junction boxes) is
+       likely to have 3x2 blocks. The number of blocks along the longest
+       side of the module is 2 and along the shortest side is 3.
+    4. A module without bypass diodes doesn't constitute a block, but may be
+       part of one.
+
+    Examples
+    --------
+    Minimal example. For a complete example, see
+    `this gallery example <sphx_glr_gallery_plot_martinez_shade_loss.py>`_.
+
+    >>> import numpy as np
+    >>> from pvlib import shading
+    >>> total_blocks = 3  # blocks along the vertical of the module
+    >>> POA_direct = 600  # W
+    >>> POA_diffuse = 80  # W
+    >>> shaded_fraction = shading.shaded_fraction1d(
+    >>>     80, 180, 90, 25,
+    >>>     collector_width=0.5, pitch=1, surface_to_axis_offset=0,
+    >>>     cross_axis_slope=5.711, shading_row_rotation=50)
+    >>> shaded_blocks = np.ceil(total_blocks*shaded_fraction)
+    >>> loss_correction = shading.direct_martinez(
+    >>>     shaded_fraction, shaded_blocks, total_blocks)
+    >>> POA_total = POA_direct * loss_correction + POA_diffuse
```

Comment:
```suggestion
    >>> import numpy as np
    >>> from pvlib import shading
    >>> total_blocks = 3  # blocks along the vertical of the module
    >>> POA_direct = 600  # W
    >>> POA_diffuse = 80  # W
    >>> # calculation of the shaded fraction for the collector
    >>> shaded_fraction = shading.shaded_fraction1d(
    >>>     solar_zenith=80, solar_azimuth=180,
     >>>     axis_azimuth=90, shaded_row_rotation=25,
    >>>     collector_width=0.5, pitch=1, surface_to_axis_offset=0,
    >>>     cross_axis_slope=5.711, shading_row_rotation=50)
    >>> # calculation of the number of shaded blocks 
    >>> shaded_blocks = np.ceil(total_blocks*shaded_fraction)
    >>> # apply the Martinez correction to the calculated shading
    >>> loss_correction = shading.direct_martinez(
    >>>     shaded_fraction, shaded_blocks, total_blocks)
    >>> # calculation of the POA for the corrected direct irradiance
    >>> POA_total = POA_direct * loss_correction + POA_diffuse
```

Reply:
I know it is a minimal example, but still I think it is nice to give some info on the steps (just added some comments, feel free to change)